### PR TITLE
more granular error reporting in Maintenance

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Add more granular error reporting to `reportInCurrent` maintenance function, and add lots of assertions.
This is not a bugfix, but a vehicle to simplify debugging and bug hunts.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13495/

### CODE REVIEW

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -496,6 +496,7 @@ VPackBuilder getShardMap(VPackSlice const& collections) {
     // But then shardMap can also be empty, so we are good.
     if (collections.isObject()) {
       for (auto collection : VPackObjectIterator(collections)) {
+        TRI_ASSERT(collection.value.isObject());
```
Comment (kvahed) [inline arangod/Cluster/Maintenance.cpp:499]:
```suggestion
        TRI_ASSERT(collection.value.isObject());
        if (collection.value.hasKey(SHARDS) && collection.value.get(SHARDS).isObject()) {
```

File: arangod/Cluster/Maintenance.cpp [hunk 2]
```diff
@@ -1121,12 +1124,15 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
       pdb = pit->second->slice()[0];
       std::vector<std::string> ppath{
         AgencyCommHelper::path(), PLAN, COLLECTIONS, dbName};
-      if (pdb.hasKey(ppath)) {   // Plan of this database's collections
-        pdb = pdb.get(ppath);
+      TRI_ASSERT(pdb.isObject());
+
+      TRI_ASSERT(pdb.isObject());
+
```
Comment (kvahed) [inline arangod/Cluster/Maintenance.cpp:1130]: Twice `TRI_ASSERT(pdb.isObject());`?

**Review by kvahed — CHANGES_REQUESTED** [review]
smaller changes

File: arangod/Cluster/Maintenance.cpp (hunk 1 above)
Comment (kvahed) [reply arangod/Cluster/Maintenance.cpp:499]: Needs closing braket of course.

File: arangod/Cluster/Maintenance.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/Cluster/Maintenance.cpp:1130]: Indeed that is too many...

Comment (jsteemann) [inline arangod/Cluster/Maintenance.cpp:1130]:
```suggestion
```

File: arangod/Cluster/Maintenance.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Cluster/Maintenance.cpp:499]: Applied the suggestion, slightly differently.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13498/

**Review by kvahed — APPROVED** [review]
LGTM

**Pull request merged**